### PR TITLE
Change stats update params to contain a list.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/38/
Test FAILed.

File: csclient/params/params.go [hunk 1]
```diff
@@ -54,6 +54,12 @@ const (
...
 // See https://github.com/juju/charmstore/blob/v4/docs/API.md#stats-update
 type StatsUpdateRequest struct {
+	Entries []StatsUpdateEntry
+}
+
+// StatsUpdateEntry holds an entry of the StatsUpdateRequest for a put to /stats/update.
+// See https://github.com/juju/charmstore/blob/v4/docs/API.md#stats-update
+type StatsUpdateEntry struct {
 	Timestamp      time.Time
 	Type           string
```
Comment (rogpeppe) [inline csclient/params/params.go]: It'd be quite nice to have a comment saying what the type is here (and what its possible values are).

**Comment by rogpeppe** [comment]
LGTM with one trivial thought.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/39/
Test FAILed.

File: csclient/params/params.go [hunk 2]
```diff
@@ -54,9 +54,15 @@ const (
...
-	Type           string
-	CharmReference *charm.Reference
+	Entries []StatsUpdateEntry
+}
+
+// StatsUpdateEntry holds an entry of the StatsUpdateRequest for a put to /stats/update.
+// See https://github.com/juju/charmstore/blob/v4/docs/API.md#stats-update
+type StatsUpdateEntry struct {
+	Timestamp      time.Time        // Time when the update did happen.
+	Type           string           // One of the constant Download, Traffic or Deploy.
```
Comment (rogpeppe) [inline csclient/params/params.go]: Maybe define this as a type:

e.g.

    type StatsUpdateType string
    const (
        UpdateDownload StatsUpdateType = "download"
        UpdateTraffic StatsUpdateType = "traffic"
        UpdateDeploy StatsUpdateType = "deploy"
     )

?

Then the possible values become more obvious and are easily commented themselves, etc.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/40/
Test FAILed.

**Pull request merged**